app/Traints/FormRequest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Traints;
+
+use App\Models\TipoPessoa;
+use App\Models\User;
+use Illuminate\Foundation\Http\FormRequest as FormRequestBase;
+use Illuminate\Support\Str;
+
+trait FormRequest
+{
+    /**
+     * Retorna os tipos de pessoa
+     *
+     * @return string
+     */
+    protected function getTiposPessoa(): string
+    {
+        return implode(',', array_keys(TipoPessoa::TIPO_PESSOA));
+    }
+
+    /**
+     * Retorna os tipos de status
+     *
+     * @return string
+     */
+    protected function getStatus(): string
+    {
+        return implode(',', array_keys(User::STATUS_USUARIO));
+    }
+
+    /**
+     * Valida, remove caracteres e faz o merge do CPF ou CNPJ.
+     *
+     * @param FormRequestBase $formRequest
+     */
+    protected function mergeExistsCpfOrCnpj(FormRequestBase $formRequest): void
+    {
+        $attributes = [];
+        if (isset($formRequest->cpf)) {
+            $attributes['cpf'] = preg_replace('/[^0-9]/', '', $formRequest->cpf);
+        }
+
+        if (isset($formRequest->cnpj)) {
+            $attributes['cnpj'] = preg_replace('/[^0-9]/', '', $formRequest->cnpj);
+        }
+
+        $formRequest->merge($attributes);
+    }
+
+    /**
+     * Valida, transforma o caracter de maúscula e faz o merge no status.
+     *
+     * @param FormRequestBase $formRequest
+     */
+    protected function mergeExistsStatus(FormRequestBase $formRequest): void
+    {
+        if (isset($formRequest->status)) {
+            $formRequest->merge([
+                'status' => Str::upper($formRequest->status)
+            ]);
+        }
+    }
+
+    /**
+     * Valida se o e-mail pertence ao Id.
+     *
+     * @param FormRequestBase $formRequest
+     * @return bool
+     */
+    protected function validateEmailBelongsToId(FormRequestBase $formRequest): bool
+    {
+        $usuario = User::whereEmail($formRequest->email)
+            ->find($formRequest->id);
+
+        return $usuario->exists ?? false;
```

Comment:
Esse fluxo tem risco de gerar um bug, porque a variável `$usuario` pode ser NULL se não for encontrado um usuário no banco com o email igual ao que foi fornecido. Nesse cenário, `$usuario->exists` levanta uma exceção.

Para evitar problemas, sugiro verificar se `$usuario` não é igual a null antes de chamar o atributo `exists`.

Uma dica também: toda vez que você encontrar um trecho de código mais ou menos assim:
```
if ($x == $y) return true
else return false
```
Você pode substituir por:
```
return $x == $y;
```
No meu entendimento, `$usuario->exists` retorna um booleano, certo? Nesse caso, você pode simplesmente fazer:
```
return !is_null($usuario) && $usuario->exists;
```

Reply:
O operador **??** de [coalescente](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op) resolve essa questão, uma vez que o resultado de `$usuario` é null e o método `exists` não existir, ele utilizará a opção `false`